src/providers/voice/twilio.js

```diff
@@ -0,0 +1,73 @@
+/* @flow */
+import fetch from '../../util/request'
+import FormData from 'form-data'
+// Types
+import type {VoiceRequestType} from '../../models/notification-request'
+
+export default class VoiceTwilioProvider {
+  id: string = 'voice-twilio-provider'
+  accountSid: string
+  apiKey: string
+
+  constructor ({accountSid, authToken}: Object) {
+    this.accountSid = accountSid
+    this.apiKey = Buffer.from(`${accountSid}:${authToken}`).toString('base64')
+  }
+
+  /*
+   * Note: 'type', 'nature', 'messageClass' are not supported.
+   */
+  async send (request: VoiceRequestType): Promise<string> {
+    const {
+      from,
+      to,
+      url,
+      method,
+      fallbackUrl,
+      fallbackMethod,
+      statusCallback,
+      statusCallbackMethod,
+      statusCallbackEvent,
+      sendDigits,
+      machineDetection,
+      machineDetectionTimeout,
+      timeout,
+      recordingChannels,
+      recordingStatusCallback,
+      recordingStatusCallbackEvent
+    } = request
```

Comment:
All these params seem a bit too specific to Twilio. Do you need any of them (apart from `from`, `to`, and `url`)?

Reply:
Yes, for the purposes of PagerTree we actually use from, to, url, fallbackUrl, statusCallback, statusCallbackEvent, machineDetection, MachineDetectionTimeout, and timeout.

If for some response its too specific and a another provider is implemented, this type can always follow the flexible  typescript definition like in src/models/provider-email.js

Reply:
Ok, I removed `statusCallbackMethod`, `recordingChannels`, `recordingStatusCallback`, and `recordingStatusCallbackEvent` for now (not all providers propose the same possibilities, so the simplest, the better :)).

Reply:
Hmmm, for this i respectfully disagree - I think having a flexible typescript definition would be best. I  can see someone else needing this functionality, and why not just have all the options available? If another vendor comes along, we could add another definition.

Reply:
Well my first priority is to ensure that people can switch their provider easily. Then I agree that "everything should be possible". Here is what I had in mind:

```js
export type VoiceRequestType = RequestMetadataType & {
  from: string,
  to: string,
  url: string,
  method?: string,
  fallbackUrl?: string,
  fallbackMethod?: string,
  statusCallback?: string,
  statusCallbackEvent?: string[],
  sendDigits?: string,
  machineDetection?: string,
  machineDetectionTimeout?: number,
  timeout?: number,
  twilioSpecifc?: {
    statusCallbackMethod?: string,
    recordingChannels?: string,
    recordingStatusCallback?: string,
    recordingStatusCallbackEvent?: string[]
  }
}
```

What do you think?